internal/storage/storage_handle.go

```diff
@@ -124,10 +142,13 @@ func createHTTPClientHandle(ctx context.Context, clientConfig *storageutil.Stora
 // http and gRPC client.
 func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClientConfig) (sh StorageHandle, err error) {
 	var sc *storage.Client
+	// The default protocol for the Go Storage control client's folders API is gRPC.
+	// gcsfuse will initially mirror this behavior due to the client's lack of HTTP support.
+	var controlClient *control.StorageControlClient
 	if clientConfig.ClientProtocol == mountpkg.GRPC {
-		sc, err = createGRPCClientHandle(ctx, &clientConfig)
+		sc, controlClient, err = createGRPCClientHandle(ctx, &clientConfig)
```

Comment:
this should not be under same client protocol check, this if-else block will remain unaltered and a new block can be added which should create grpc control client irrespective of client protocol if enable hns flag is true